src/components/Search/CustomSearchComponents.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  RefinementListProps,
+  SortByProps,
+  useRefinementList,
+  useSortBy,
+} from 'react-instantsearch-hooks-web';
+import { Accordion, Chip, Group, MultiSelect, Select, Text } from '@mantine/core';
+import { useEffect, useState } from 'react';
+import { useDebouncedValue } from '@mantine/hooks';
+
+export function SortBy({ title, ...props }: SortByProps & { title: string }) {
+  const { options, refine, currentRefinement, ...args } = useSortBy(props);
+
+  if (options.length === 0) {
+    return null;
+  }
+
+  return (
+    <Accordion defaultValue={title} variant="filled">
+      <Accordion.Item value={title}>
+        <Accordion.Control>
+          <Text size="md" weight={500}>
+            {title}
+          </Text>
+        </Accordion.Control>
+        <Accordion.Panel>
+          <Select
+            name="sort"
+            data={options}
+            value={currentRefinement}
+            onChange={(value) => refine(value || options[0].value)}
+          />
+        </Accordion.Panel>
+      </Accordion.Item>
+    </Accordion>
+  );
+}
+
+export function SearchableMultiSelectRefinementList({
+  title,
+  ...props
+}: RefinementListProps & { title: string }) {
+  const { items, refine, searchForItems, isFromSearch } = useRefinementList({ ...props });
+  const [searchValue, setSearchValue] = useState('');
+  const [debouncedSearchValue] = useDebouncedValue(searchValue, 300);
+  // We need to keep the state of the select here because the items may dissapear while searching.
+  const [refinedItems, setRefinedItems] = useState<typeof items>([]);
```

Comment:
Wouldn't it be good to initialize this with the `items`?
```suggestion
  const [refinedItems, setRefinedItems] = useState<typeof items>(items ?? []);
```
Not sure if items is null at some point, so doing fallback to empty array

Reply:
Items is not the same as refined items tho